packages/fx-core/tests/plugins/solution/debug/solution.debug.scaffolding.test.ts

```diff
@@ -327,6 +364,42 @@ describe("solution.debug.scaffolding", () => {
 
         await assertLocalDebugLocalEnvs(v2Context, inputs, parameter.numLocalEnvs);
       });
+
+      it(`happy path: tab without function and bot (${parameter.programmingLanguage}) and AAD`, async () => {
+        const projectSetting = {
+          appName: "",
+          projectId: uuid.v4(),
+          solutionSettings: {
+            name: "",
+            version: "",
+            hostType: "Azure",
+            capabilities: ["Tab", "Bot"],
+            activeResourcePlugins: [],
+          },
+          programmingLanguage: parameter.programmingLanguage,
+        };
+        const v2Context = new MockedV2Context(projectSetting);
+        const result = await scaffoldLocalDebugSettings(v2Context, inputs);
+        chai.assert.isTrue(result.isOk());
+
+        //assert output launch.json
+        const launch = fs.readJSONSync(expectedLaunchFile);
+        const configurations: [] = launch["configurations"];
+        const compounds: [] = launch["compounds"];
+        chai.assert.equal(configurations.length, parameter.numConfigurations);
+        chai.assert.equal(compounds.length, parameter.numCompounds);
+
+        //assert output tasks.json
+        const tasksAll = fs.readJSONSync(expectedTasksFile);
+        const tasks: [] = tasksAll["tasks"];
+        chai.assert.equal(tasks.length, parameter.numTasks);
+
+        //assert output settings.json
+        const settings = fs.readJSONSync(expectedSettingsFile);
+        chai.assert.equal(Object.keys(settings).length, 1);
+
+        await assertLocalDebugLocalEnvs(v2Context, inputs, parameter.numLocalEnvs - 12);
```

Comment:
add comments about why we need to `-12`?
By the way, can we define expected content for non-aad situations? So we don't need to have magic numbers here.

Reply:
Fixed. Thanks!